plugin-api/src/main/java/org/hyperledger/besu/plugin/services/TransactionSelectionService.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.besu.plugin.services;
+
+import org.hyperledger.besu.plugin.services.txselection.TransactionSelectorFactory;
+
+import java.util.Optional;
+
+public interface TransactionSelectionService extends BesuService {
+
+  Optional<TransactionSelectorFactory> get();
+
+  void registerTransactionSeclectorFactory(TransactionSelectorFactory transactionSelectorFactory);
```

Comment:
typo here

Reply:
Do you mean the "selector" part? I did that on purpose, because the service is a "selection" service, but the two other classes are the "selector" and the factory for the selector. Does that make sense? I'm not married to the names :-)

Reply:
```suggestion
  void registerTransactionSelectorFactory(TransactionSelectorFactory transactionSelectorFactory);
```